### PR TITLE
Ensure ntp is installed (when Install Recommended Packages is off)

### PR DESCRIPTION
**Pull request by mvidner**
https://bugzilla.suse.com/show_bug.cgi?id=938951

The actual fix is https://github.com/yast/yast-country/commit/13e61a9e8d5f60d8ecacb301b4429a36e1c7a2c3 and the rest is refactoring. But the code is spread across two packages (see a related PR in ntp-client) and I am not sure how much of a cleanup I should attempt now (and even if this is in the end a cleanup)

### CODE REVIEW

File: timezone/src/lib/timezone/ntp.rb [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+
+    # This proxy handles the complication
+    # that the package yast2-ntp-client may not be present
+    # (on the running system).
+    # The called client has been meanwhile moved to this package but it
+    # does not check for the dependencies itself yet.
+    def ntp_call(acall, args)
+      if !@installed
+        # replace "replace_point" by the widgets
+        if acall == "ui_init"
```
Comment (lslezak) [inline timezone/src/lib/timezone/ntp.rb]: NP: I know this was simply copied from elsewhere, but using `case...when` here would really improve it a lot...

File: timezone/src/lib/timezone/ntp_adapter.rb [hunk 2]
```diff
@@ -0,0 +1,476 @@
...
+        @func,
+        @param
+      )
+
+      # FIXME must go to module to preserve value
+      @ntp_was_used = false
+
+
+
+      if false
```
Comment (lslezak) [inline timezone/src/lib/timezone/ntp_adapter.rb]: NP: This `if false` is useless, and the rest could be replaced by `case...when` as well...

File: timezone/src/lib/timezone/ntp_adapter.rb [hunk 3]
```diff
@@ -0,0 +1,476 @@
...
+              required_package
+            )
+          )
+        end
+      end
+
+      ret = 0
+      if NetworkService.isNetworkRunning
+        # Only if network is running try to synchronize the ntp server
+        Popup.ShowFeedback("", _("Synchronizing with NTP server..."))
```
Comment (lslezak) [inline timezone/src/lib/timezone/ntp_adapter.rb]: NP: The new `Popup.Feedback` with a block should be used here...

**Comment by ancorgs** [comment]
To be honest, such a big refactoring (although a big part is just code moving) without unit test scares me a bit.

**Comment by mvidner** [comment]
@ancorgs You're right, and the API redesign is not too good either; so I am going to improve that and add tests, but proceed with the actual simple fix in another PR.

**Comment by jreidinger** [comment]
@mvidner what is status here? I worry it is too risky for SP1 now, but if you improve it a bit, it can be merged for SP2

**Comment by kobliha** [comment]
@mvidner SP2 is just behind the door, any progress?

**Comment by teclator** [comment]
There has not been updates since a while and the bug was already fixed by https://github.com/yast/yast-ntp-client/pull/34 so maybe make sense to close it by now and just create a new PR if some improvements are done at the end.

@mvidner , what do you think about?

**Pull request closed without merging**